tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx

```diff
@@ -1343,21 +1344,30 @@ void SetBranchesHelper(TTree *inputTree, TTree &outputTree, const std::string &i
       }
    }
    auto *outputBranch = outputBranches.Get(outName);
-   const bool isTClonesArray = inputBranch != nullptr && std::string(inputBranch->GetClassName()) == "TClonesArray";
-   const auto mustWriteRVec = !inputBranch || isTClonesArray ||
-                              ROOT::ESTLType::kSTLvector == TClassEdit::IsSTLCont(inputBranch->GetClassName());
+
+   // if no backing input branch, we must write out an RVec
+   bool mustWriteRVec = (inputBranch == nullptr);
+   // otherwise, if input branch is TClonesArray, must write out an RVec
+   if (!mustWriteRVec && std::string_view(inputBranch->GetClassName()) == "TClonesArray") {
+      mustWriteRVec = true;
+      Warning("Snapshot",
```

Comment:
Do you have a test for this case?

Reply:
yes, it's here: https://github.com/root-project/root/blob/master/tree/dataframe/test/dataframe_snapshot.cxx#L782 (I'm not introducing this warning in this PR)